engines/router/missionctl/handlers/batch_http_handler_test.go

```diff
@@ -0,0 +1,171 @@
+			expectedContentType:  "text/plain; charset=utf-8",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			req := httptest.NewRequest(http.MethodPost, "/v1/batch_predict", bytes.NewBuffer([]byte(test.payload)))
+			w := httptest.NewRecorder()
+			batchHTTPHandler.ServeHTTP(w, req)
+			res := w.Result()
+			defer res.Body.Close()
+			result, _ := ioutil.ReadAll(res.Body)
+
+			assert.Equal(t, test.expectedStatusCode, res.StatusCode)
+			assert.Equal(t, test.expectedContentType, res.Header.Get("Content-Type"))
+			if test.expectedContentType == "application/json" {
+				assert.JSONEq(t, test.expectedResponseBody, string(result))
+			} else {
+				assert.Equal(t, test.expectedResponseBody, string(result))
+			}
+		})
+	}
+}
+
+func TestNewBatchHTTPHandlerBadRoute(t *testing.T) {
+	missionCtl := &MockMissionControlBadRoute{BaseMockMissionControl: *createTestBaseMissionControl()}
+	batchHTTPHandler := NewBatchHTTPHandler(missionCtl)
+	assert.NotNil(t, batchHTTPHandler)
+
+	tests := map[string]struct {
+		payload              string
+		expectedResponseBody string
+		expectedStatusCode   int
+		expectedContentType  string
+	}{
+		"bad route request": {
+			payload: `[
+							{ "request1": "value1" },
+							{ "request2": "value2" }
+					  ]`,
+			expectedResponseBody: `[
+										{ "code": 500, 
+										  "error": "Bad Route Called"
+										},
+										{ "code": 500, 
+										  "error": "Bad Route Called"
+										}
+								   ]`,
+			expectedStatusCode:  200,
+			expectedContentType: "application/json",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			req := httptest.NewRequest(http.MethodPost, "/v1/batch_predict", bytes.NewBuffer([]byte(test.payload)))
+			w := httptest.NewRecorder()
+			batchHTTPHandler.ServeHTTP(w, req)
+			res := w.Result()
+			defer res.Body.Close()
+			result, _ := ioutil.ReadAll(res.Body)
+
+			assert.Equal(t, test.expectedStatusCode, res.StatusCode)
+			assert.Equal(t, test.expectedContentType, res.Header.Get("Content-Type"))
+			assert.JSONEq(t, test.expectedResponseBody, string(result))
+		})
+	}
+}
+
+func createRouteServer(t *testing.T, testRouterHandler http.HandlerFunc) *httptest.Server {
+	mux := http.NewServeMux()
+	mux.Handle("/route1", testRouterHandler)
+	server := httptest.NewUnstartedServer(mux)
+	listener, err := net.Listen("tcp", "127.0.0.1:8080")
+	if err != nil {
+		t.Fatal("Failed to start test http server: " + err.Error())
+	}
+	server.Listener = listener
+	return server
+}
+
+func createGenericMissionControl(t *testing.T) missionctl.MissionControl {
+	//Create missionctl with route for testing
+	missionCtl, err := missionctl.NewMissionControl(
+		nil,
+		&config.EnrichmentConfig{
+			Endpoint: "",
+			Timeout:  time.Second,
+		},
+		&config.RouterConfig{
+			//ConfigFile: filepath.Join("../testdata", "nop_default_router.yaml"),
+			ConfigFile: filepath.Join("../testdata", "batch_router_test.yaml"),
```

Comment:
Maybe this ConfigFile can be passed as an argument to `createGenericMissionControl` function?